nixos/modules/services/networking/openvpn.nix

```diff
@@ -53,11 +53,15 @@ let
         ${optionalString (
           cfg.down != "" || cfg.updateResolvConf
         ) "down ${pkgs.writeShellScript "openvpn-${name}-down" downScript}"}
-        ${optionalString (cfg.authUserPass != null)
-          "auth-user-pass ${pkgs.writeText "openvpn-credentials-${name}" ''
-            ${cfg.authUserPass.username}
-            ${cfg.authUserPass.password}
-          ''}"
+        ${lib.optionalString (cfg.authUserPass != null) (
+          if lib.isAttrs cfg.authUserPass then
```

Comment:
`else` is now expected to be a string (absolute file path) to be included verbatim in the generated config. I mention this because I'm not sure if I should have instead extracted `authUserPassType` and called `authUserPassType.check` instead of `isAttrs`, but I think the usage is simple enough that the latter is fine for the foreseeable future. I welcome opinions.